operator/internal/manifests/storage/schema.go

```diff
@@ -0,0 +1,71 @@
+package storage
+
+import (
+	"time"
+
+	lokiv1beta1 "github.com/grafana/loki/operator/api/v1beta1"
+)
+
+const (
+	// DaysToUpdate is the number of days before a schema
+	// update will be applied to a cluster.
+	DaysToUpdate = 2
+)
+
+// UpdateSchemas returns a list of schemas that can be used by Loki. It will append a new
+// schema if there are no existing, the last schema is not the latest, or the last schema
+// is not using the right object storage.
+func UpdateSchemas(version lokiv1beta1.ObjectStorageSchemaVersion, existing []Schema) []Schema {
+	count := len(existing)
+	date := time.Now().UTC()
+
+	if count == 0 {
+		return []Schema{
+			{
+				From:    date.Format("2006-01-02"),
+				Version: version,
+			},
+		}
+	}
+
+	last := existing[count-1]
+
+	// Apply the changes to the index for logs starting two days later.
+	// Since the timestamps work by UTC, the goal is to avoid the scenario
+	// in which logs might be inaccessible because they were created with the
+	// wrong schema.
+	nextAppliedDate := date.Add(time.Hour * 24 * DaysToUpdate).Format("2006-01-02")
+
+	if last.Version != version {
+		// There is a pending update that has not been applied yet.
+		// In order to avoid possible timing scenarios, the following
+		// case is not in consideration:
+		//
+		// Day 0: Accept and set update to v12 for two days in the future
+		// Day 1: Accept and set update to v11 (v12 may not have been applied yet)
+		// Day 2: v11 Update will be applied
+		//
+		// Since the UTC timing is important and to avoid risk of multiple
+		// updates at wrong intervals, the v12 update would be applied on Day 2
+		// and the v11 update would be applied on Day 3. However, changes on the same
+		// day (as implied by the date being the same) should be fine.
```

Comment:
Is this doc comment valid? I don't quite get the "case is not in consideration" vs the code below. In addition a `Day 3` is mentioned. Maybe we use case use example dates to enhance comprehension.

Reply:
Yes, adding dates for clarity make sense.

With the idea of "downgrading" the schema version, there is a case which could happen:

2022-4-27: Apply change to v12, this will be landed on 2022-4-29
2022-4-28: Apply change to v11, this will be landed on 2022-4-30

If this change happened early enough, the operator could have removed that v12 change in place of the v11 one. To keep things simple, I opted to just apply both changes on their given days rather than attempt to modify the v12 change to v11. Since there is already a two day gap to try and avoid timing troubles, it didn't make sense.